controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -1232,6 +1234,55 @@ func TestKubeadmControlPlaneReconciler_reconcileDelete(t *testing.T) {
 		g.Expect(controlPlaneMachines.Items).To(HaveLen(3))
 	})
 
+	t.Run("does not remove any control plane Machines if MachinePools exist", func(t *testing.T) {
+		_ = feature.MutableGates.Set("MachinePool=true")
+		g := NewWithT(t)
+
+		cluster, kcp, _ := createClusterWithControlPlane()
+		controllerutil.AddFinalizer(kcp, controlplanev1.KubeadmControlPlaneFinalizer)
+
+		workerMachinePool := &expv1.MachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "worker",
+				Namespace: cluster.Namespace,
+				Labels: map[string]string{
+					clusterv1.ClusterLabelName: cluster.Name,
+				},
+			},
+		}
+
+		initObjs := []client.Object{cluster.DeepCopy(), kcp.DeepCopy(), workerMachinePool.DeepCopy()}
```

Comment:
nit: This deepCopy is fine though is a bit redundant. I see this is calling `newFakeClient` which calls `NewFakeClientWithScheme` which already deepCopy the objects https://github.com/kubernetes-sigs/cluster-api/blob/58583914ed0034203ee398f864b5e023b2ed65e2/test/helpers/client.go#L38

Also the fakeClient tracker itself deepCopy as well.

https://github.com/kubernetes/client-go/blob/master/testing/fixture.go#L373-L376